pkgs/build-support/rust/build-rust-package/default.nix

```diff
@@ -45,7 +45,7 @@
 , checkFeatures ? buildFeatures
 , useNextest ? false
 # Enable except on aarch64 pkgsStatic, where we use lld for reasons
-, auditable ? !cargo-auditable.meta.broken && !(stdenv.hostPlatform.isStatic && stdenv.hostPlatform.isAarch64 && !stdenv.hostPlatform.isDarwin)
+, auditable ? !cargo-auditable.meta.broken && (!stdenv.hostPlatform.isStatic || !stdenv.hostPlatform.isAarch64 || stdenv.hostPlatform.isDarwin) && !stdenv.hostPlatform.isWasi
```

Comment:
I think the motivation for the logic here is more difficult to understand with the ! distributed inside.  Why change that?

Would probably also be clearer to move the wasi case before the complicated one with the parentheses, and maybe start a new line after each top-level `&&`.

Reply:
I wasn't able to wrap my head around what I wanted to do without putting this into negation normal form (personal preference? double negation is mean). The motivation is to disable auditable in "this *or* that" situation, yet the top level operator is an &&.

I considered doing this

```diff
diff --git a/pkgs/build-support/rust/build-rust-package/default.nix b/pkgs/build-support/rust/build-rust-package/default.nix
index 36e0408cc198..d85820ab536a 100644
--- a/pkgs/build-support/rust/build-rust-package/default.nix
+++ b/pkgs/build-support/rust/build-rust-package/default.nix
@@ -44,9 +44,14 @@
 , buildFeatures ? [ ]
 , checkFeatures ? buildFeatures
 , useNextest ? false
-# Enable except on aarch64 pkgsStatic, where we use lld for reasons
-, auditable ? !cargo-auditable.meta.broken && !(stdenv.hostPlatform.isStatic && stdenv.hostPlatform.isAarch64 && !stdenv.hostPlatform.isDarwin)
-
+, auditable ? !( # rust binaries are made auditable except for any of these:
+    # - cargo-auditable is broken
+    cargo-auditable.meta.broken
+    # - we're on wasi
+    || stdenv.hostPlatform.isWasi
+    # - were on static aarch64 except darwin
+    || (stdenv.hostPlatform.isStatic && stdenv.hostPlatform.isAarch64 && !stdenv.hostPlatform.isDarwin)
+  )
 , depsExtraArgs ? {}

 # Toggles whether a custom sysroot is created when the target is a .json file.
 ```

but not doing this at all just causes a little warning at the end of the build

```
WARNING: target 'wasm32-wasip1' is not supported by 'cargo auditable'!
The build will continue, but no audit data will be injected into the binary.
```

Ignoring that warning and not touching this line at all seems like the lesser evil.

Reply:
Agreed.